Doc/library/pprint.rst

```diff
@@ -183,6 +183,10 @@ One more support function is also defined:
    recursive reference will be represented as ``<Recursion on typename with
    id=number>``.  The representation is not otherwise formatted.
 
+   Note that for recursive protection to apply, data structures have to be
+   subclassed from :class:`~object.dict`, :class:`~object.list` or
+   :class:`~object.tuple` and the ``__repr__`` method must not be overridden.
+
```

Comment:
Maybe we can merge this with the previous paragraph, like:

Return a string representation of *object*, protected against recursion in some common data
   structures, namely :class:`~object.dict`, :class:`~object.list` and  :class:`~object.tuple` whose
``__repr__`` has not been overridden.  If the representation of *object* exposes a recursive entry, the
    recursive reference will be represented as ``<Recursion on typename with  id=number>``.  The
representation is not otherwise formatted.


Does that read better?

Reply:
Sounds good to me, pushed the change. Also I noticed that isrecursive() is subject to same limitations, so I've added a short note referencing `saferepr()`.